### PR TITLE
feat(builder): add option to include source map in plugin bundles

### PR DESCRIPTION
**Pull request by kiancross**
**What's the problem this PR addresses?**

Plugins are sometimes difficult to debug because they do not include source maps. This PR adds an option to include source maps in the bundle.

Note that the plugin developer must include [`source-map-support`](https://www.npmjs.com/package/source-map-support) with the bundle for any stack-traces produced by `yarn` to be correct.
...

**How did you fix it?**

* Added an additional CLI option.
* Replaced `webpack.sources.RawSource` with `ConcatSource`, which supports source maps.

...

**Checklist**

- [x] I have read the [Contributing Guide](https://yarnpkg.com/advanced/contributing).

- [x] I have set the packages that need to be released for my changes to be effective.

- [x] I will check that all automated PR checks pass before the PR gets reviewed.

### CODE REVIEW

File: packages/yarnpkg-builder/sources/commands/build/plugin.ts [hunk 1]
```diff
@@ -108,18 +117,20 @@ export default class BuildPluginCommand extends Command {
...
             {apply: (compiler: webpack.Compiler) => {
-              compiler.hooks.compilation.tap(`MyPlugin`, (compilation: webpack.Compilation) => {
-                compilation.hooks.optimizeChunkAssets.tap(`MyPlugin`, (chunks: Set<webpack.Chunk>) => {
+              compiler.hooks.compilation.tap(`WrapperPlugin`, (compilation: webpack.Compilation) => {
+                compilation.hooks.optimizeChunkAssets.tap(`WrapperPlugin`, (chunks: Set<webpack.Chunk>) => {
                   for (const chunk of chunks) {
                     for (const file of chunk.files) {
                       // @ts-expect-error
-                      compilation.assets[file] = new webpack.sources.RawSource(
+                      compilation.assets[file] = new ConcatSource(
```
Comment (merceyz) [inline packages/yarnpkg-builder/sources/commands/build/plugin.ts:125]:
```suggestion
                      compilation.assets[file] = new webpack.sources.ConcatSource(
```

Don't need to add the dependency for this as webpack already exports it

**Review by merceyz — CHANGES_REQUESTED** [review]
We should probably just support users customizing their configs, as there might be other, specific stuff, that others need

File: packages/yarnpkg-builder/sources/commands/build/plugin.ts (hunk 1 above)
Comment (kiancross) [reply packages/yarnpkg-builder/sources/commands/build/plugin.ts:125]: > Don't need to add the dependency for this as webpack already exports it.

Ah ok - will remove.

> We should probably just support users customizing their configs, as there might be other, specific stuff, that people might need

Agreed.

File: .yarn/versions/a610ce95.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+releases:
+  "@yarnpkg/builder": minor
```
Comment (merceyz) [inline .yarn/versions/a610ce95.yml:2]: Only need to release the builder, the rest can be declined

File: packages/yarnpkg-builder/sources/commands/build/bundle.ts [hunk 3]
```diff
@@ -43,6 +43,9 @@ export default class BuildBundleCommand extends Command {
   @Command.Boolean(`--no-minify`, {description: `Build a bundle for development, without optimizations (minifying, mangling, treeshaking)`})
   noMinify: boolean = false;
 
+  @Command.Boolean(`--include-source-map`, {description: `Includes a source map in the bundle`})
+  includeSourceMap: boolean = false;
```
Comment (arcanis) [inline packages/yarnpkg-builder/sources/commands/build/bundle.ts:47]: I'll just rename that into `--source-map`, which I think is more in line with the general option style.

File: packages/yarnpkg-builder/sources/commands/build/plugin.ts [hunk 4]
```diff
@@ -6,6 +6,7 @@ import path                                                                 from
 import TerserPlugin                                                         from 'terser-webpack-plugin';
 import webpack                                                              from 'webpack';
 
+
```
Comment (merceyz) [inline packages/yarnpkg-builder/sources/commands/build/plugin.ts:9]:
```suggestion
```

**Review by arcanis — APPROVED** [review]

**Review by merceyz — APPROVED** [review]

**Pull request merged**